gpcontrib/gpcloud/include/s3common_headers.h

```diff
@@ -26,6 +26,23 @@ using std::string;
 using std::stringstream;
 using std::vector;
 
+#define DO_PRAGMA(x) _Pragma (#x)
+#if defined(__clang__)
+	#define SUPPRESS_COMPILER_WARNING(expr, warning) do { \
+		_Pragma("clang diagnostic push") \
+		DO_PRAGMA(clang diagnostic ignored warning) \
+		expr; \
+		_Pragma("clang diagnostic pop") \
+	} while(0)
```

Comment:
Since it is only used with "-Wdeprecated-declarations", can this just be changed to something like this:
```suggestion
	#define SUPPRESS_DEPRECATION(expr) do { \
		_Pragma("clang diagnostic push") \
		DO_PRAGMA(clang diagnostic ignored "-Wdeprecated-declarations") \
		expr; \
		_Pragma("clang diagnostic pop") \
	} while(0)
```

Can this technique be used to push, include the header with deprecated functions, and then pop to avoid the need to use this macro for every call?

Reply:
> Since it is only used with "-Wdeprecated-declarations", can this just be changed

No, more warnings in the future might appear and they won't necessarily be like this one. This macro was made as generic as possibly intentionally.

> Can this technique be used to push, include the header with deprecated functions, and then pop to avoid the need to use this macro for every call?

What do you mean? We either use a macro for every line which needs to be ignored or just ignore everything in a file. Why push-pop for each line? Also, I don't want to use push-pop semantics as one can forget to "pop" and ignore warnings which must not be ignored. One-liners make it more clear which expression should be ignored as well.

Reply:
> No, more warnings in the future might appear

Do we plan to introduce more warnings? :D

Reply:
Can you control third-party code?